src/core/Akka.Remote.Tests/Transport/HeliosTransportDnsResolutionSpec.cs

```diff
@@ -0,0 +1,190 @@
+        }
+
+        public static string ParseAddress(EndPoint endpoint)
+        {
+            if (endpoint is IPEndPoint)
+                return ((IPEndPoint)endpoint).Address.ToString();
+            return ((DnsEndPoint)endpoint).Host;
+        }
+    }
+
+
+    /// <summary>
+    /// Designed to guarantee that the default Akka.Remote transport "does the right thing" with respect
+    /// to DNS resolution and IP binding under a variety of scenarios
+    /// </summary>
+    public class HeliosTransportDnsResolutionSpec : AkkaSpec
+    {
+        public HeliosTransportDnsResolutionSpec()
+        {
+            Arb.Register(typeof(EndpointGenerators));
+        }
+
+        public Config BuildConfig(string hostname, int? port = null, string publichostname = null)
+        {
+            return ConfigurationFactory.ParseString(@"akka.actor.provider = ""Akka.Remote.RemoteActorRefProvider, Akka.Remote""")
+                .WithFallback("akka.remote.helios.tcp.hostname =\"" + hostname + "\"")
+                .WithFallback("akka.remote.helios.tcp.public-hostname =\"" + (publichostname ?? hostname) + "\"")
+                .WithFallback("akka.remote.helios.tcp.port = " + (port ?? 0))
+                .WithFallback("akka.test.single-expect-default = 1s")
+                .WithFallback(Sys.Settings.Config);
+        }
+
+        private class AssociationAcker : ReceiveActor
+        {
+            public AssociationAcker()
+            {
+                ReceiveAny(o => Sender.Tell("ack"));
+            }
+        }
+
+        private void Setup(string inboundHostname, string outboundHostname, string inboundPublicHostname = null, string outboundPublicHostname = null)
+        {
+            _inbound = ActorSystem.Create("Sys1", BuildConfig(inboundHostname, 0, inboundPublicHostname));
+            _outbound = ActorSystem.Create("Sys2", BuildConfig(outboundHostname, 0, outboundPublicHostname));
+
+            _inbound.ActorOf(Props.Create(() => new AssociationAcker()), "ack");
+            _outbound.ActorOf(Props.Create(() => new AssociationAcker()), "ack");
+
+            var addrInbound = RARP.For(_inbound).Provider.DefaultAddress; 
+            var addrOutbound = RARP.For(_outbound).Provider.DefaultAddress; 
+
+            _inboundAck = new RootActorPath(addrInbound) / "user" / "ack";
+            _outboundAck = new RootActorPath(addrOutbound) / "user" / "ack";
+
+            _inboundProbe = CreateTestProbe(_inbound);
+            _outboundProbe = CreateTestProbe(_outbound);
+        }
+
+        private void Cleanup()
+        {
+            Shutdown(_inbound, TimeSpan.FromSeconds(1));
+            Shutdown(_outbound, TimeSpan.FromSeconds(1));
+        }
+
+        private ActorSystem _inbound;
+        private ActorSystem _outbound;
+        private ActorPath _inboundAck;
+        private ActorPath _outboundAck;
+        private TestProbe _inboundProbe;
+        private TestProbe _outboundProbe;
+
+        [Property(MaxTest = 25)]
```

Comment:
Long-running Property test, but verifies that any Akka.Remote connection using any of these valid endpoints can successfully request-response.